File: app/utils/session_manager.py
import json
from pathlib import Path
from typing import Any, Dict, Optional

SESSIONS_DIR = Path(__file__).resolve().parent.parent.parent / "sessions"
SESSIONS_DIR.mkdir(parents=True, exist_ok=True)

def _session_path(username: str) -> Path:
    return SESSIONS_DIR / f"{username}.json"

def save_session(username: str, session: Dict[str, Any]) -> None:
    _session_path(username).write_text(json.dumps(session, indent=2), encoding="utf-8")

def load_session(username: str) -> Optional[Dict[str, Any]]:
    p = _session_path(username)
    if not p.exists():
        return None
    return json.loads(p.read_text(encoding="utf-8"))

def delete_session(username: str) -> bool:
    p = _session_path(username)
    if p.exists():
        p.unlink()
        return True
    return False

def session_exists(username: str) -> bool:
    return _session_path(username).exists()

File: src/utils/sessionManager.js
const fs = require("fs");
const path = require("path");

const { IgApiClient } = require("instagram-private-api");

const SESSION_DIR = path.join(__dirname, "../../sessions");
if (!fs.existsSync(SESSION_DIR)) {
  fs.mkdirSync(SESSION_DIR, { recursive: true });
}

function getSessionPath(username) {
  return path.join(SESSION_DIR, `${username}.json`);
}

function saveSession(username, session) {
  const filePath = getSessionPath(username);
  fs.writeFileSync(filePath, JSON.stringify(session, null, 2));
}

function loadSession(username) {
  const filePath = getSessionPath(username);
  if (!fs.existsSync(filePath)) return null;
  const raw = fs.readFileSync(filePath, "utf-8");
  return JSON.parse(raw);
}

function deleteSession(username) {
  const filePath = getSessionPath(username);
  if (fs.existsSync(filePath)) {
    fs.unlinkSync(filePath);
    return true;
  }
  return false;
}

function sessionExists(username) {
  return fs.existsSync(getSessionPath(username));
}

async function getSessionClient(username) {
  console.log("Recebido username em getSessionClient:", username); // ADICIONE ESTA LINHA

  const sessionData = loadSession(username);
  if (!sessionData) throw new Error("Sessão não encontrada");

  const ig = new IgApiClient();
  ig.state.generateDevice(username);
  await ig.state.deserialize(sessionData);
  return ig;
}

module.exports = {
  saveSession,
  loadSession,
  deleteSession,
  sessionExists,
  getSessionClient,
};
